index.js

```diff
@@ -210,56 +210,90 @@ class ServerlessCustomDomain {
+    const issuedCertArn = acm.listCertificates({ CertificateStatuses: ['ISSUED'] }).promise();
 
+    return issuedCertArn.then((data) => {
+      if (process.env.SLS_DEBUG && Array.isArray(data.CertificateSummaryList)) {
+        this.serverless.cli.log(`Found ${data.CertificateSummaryList.length} Valid Certificates: ${JSON.stringify(data.CertificateSummaryList)}`);
+      }
+      return this.findClosestCertificate(data);
+    }).catch((err) => {
+      // No cert found in the ISSUED status. Let's look through certs in all other statuses.
+      const certArn = acm.listCertificates().promise();
 
-      // Checks if a certificate name is given
-      if (certificateName != null) {
-        const foundCertificate = data.CertificateSummaryList
-          .find(certificate => (certificate.DomainName === certificateName));
-
-        if (foundCertificate != null) {
-          certificateArn = foundCertificate.CertificateArn;
+      return certArn.then((data) => {
+        if (process.env.SLS_DEBUG && Array.isArray(data.CertificateSummaryList)) {
+          this.serverless.cli.log(`Found ${data.CertificateSummaryList.length} Certificates: ${JSON.stringify(data.CertificateSummaryList)}`);
         }
-      } else {
-        certificateName = this.givenDomainName;
-        data.CertificateSummaryList.forEach((certificate) => {
-          let certificateListName = certificate.DomainName;
-
-          // Looks for wild card and takes it out when checking
-          if (certificateListName[0] === '*') {
-            certificateListName = certificateListName.substr(1);
-          }
-
-          // Looks to see if the name in the list is within the given domain
-          // Also checks if the name is more specific than previous ones
-          if (certificateName.includes(certificateListName)
-            && certificateListName.length > nameLength) {
-            nameLength = certificateListName.length;
-            certificateArn = certificate.CertificateArn;
-          }
-        });
-      }
 
-      if (certificateArn == null) {
-        throw Error(`Could not find the certificate ${certificateName}`);
-      }
-      return certificateArn;
+        return this.findClosestCertificate(data);
+      }).catch(() => {
+        // rethrow the original error
+        throw err;
+      }).then((certificateArn) => {
+        const certificateName = this.serverless.service.custom.customDomain.certificateName;
+
+        // The cert was found in an invalid status. Let's get the actual status.
+        const describeCert = acm.describeCertificate({
+          CertificateArn: certificateArn,
+        }).promise();
+
+        return describeCert.then((data) => {
+          throw Error(`The certificate ${certificateName} was found with a "${data.Certificate.Status}" status but was expecting "ISSUED" status`);
+        }, () => {
+          throw Error(`The certificate ${certificateName} was found but is not in the "ISSUED" status`);
```

Comment:
What would cause the `acm.describeCertificate()` call to fail?

Reply:
IAM Permissions is normally what we run into for stuff like this.

Reply:
Ah ok I see, makes sense